src/App.tsx

```diff
@@ -1,15 +1,57 @@
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropdownMenu';
+import { PersonData } from './types/PersonData';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectPerson, setSelectPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPersonData, setSelectedPersonData] =
+    useState<PersonData | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name
+        .toLowerCase()
+        .trim()
+        .includes(appliedQuery.toLowerCase().trim()),
+    );
+  }, [appliedQuery]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
This is a naming convention issue. According to the conventions provided in the checklist, event handler functions should be named starting with 'handle' and ending with the name of the event. So, `handleInputChange` should be renamed to `handleChange`.